app/src/main/java/com/example/antitheft/MainActivity.java

```diff
@@ -0,0 +1,174 @@
+
+/**
+ * Main Activity. It takes UI commands as input, and, as an output, it gives them a certain action.
+ * Its current scope is just to send SMS, based by what does the User decide to do.
+ *
+ * @author Francesco Bau'
+ * @version 0.1
+ * @since 24/02/2020
+ */
+public class MainActivity extends AppCompatActivity {
+
+    EditText telephoneField;
+    View sendButton;
+
+    /**
+     * Context instance is not enough, it needs an Activity instance.
+     * It's static because it's used on a static method.
+     * That method is static because GPSCommandHandler needs that method, as well.
+     *
+     * @see this#sendCommand(SMSMessage)
+     */
+    private static Activity activity;
+
+    private static final String APP_CODE = "AT";
+    private static final int DEFAULT_PASSWORD = 1234;
+
+    private static final String LOCATE_COMMAND = "LOCATE";
+
+    // Full Default LOCATE command: "AT-1234 LOCATE"
+    public static final String FULL_DEFAULT_LOCATE_COMMAND = APP_CODE + "-" + DEFAULT_PASSWORD + " " + LOCATE_COMMAND;
+
+    /**
+     * Necessary permissions: SEND_SMS, RECEIVE_SMS, WAKE_LOCK,
+     * ACCESS_COARSE_LOCATION, ACCESS_FINE_LOCATION, ACCESS_BACKGROUND_LOCATION
+     */
+    static final String[] PERMISSIONS = new String[]{
+            Manifest.permission.SEND_SMS, Manifest.permission.RECEIVE_SMS, Manifest.permission.WAKE_LOCK,
+            Manifest.permission.ACCESS_COARSE_LOCATION, Manifest.permission.ACCESS_FINE_LOCATION,
+            Manifest.permission.ACCESS_BACKGROUND_LOCATION};
+
+    private static final int REQUEST_CODE = 0;
+
+    private static LocationParser locationParser = new LocationParser();
+    private static FusedLocationProviderClient client;
+
+    // If it won't be able to retrieve the location after DELAY * MAXIMUM_CHECK_TIMEOUT milliseconds,
+    // it won't wait anymore.
+    private static final int DELAY = 500;
+    private static final int MAXIMUM_CHECK_TIMEOUT = 10;
+
+    @Override
+
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_main);
+
+        Context context = getApplicationContext();
+
+        //Setting the listener
+        SMSManager.getInstance().setReceivedListener(GPSCommandReceiver.class, context);
+
+        telephoneField = findViewById(R.id.telephoneNumber);
+        sendButton = findViewById(R.id.sendButton);
+        activity = this;
+
+        sendButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                boolean permissionsGranted = checkSelfPermission(Manifest.permission.SEND_SMS) == PERMISSION_GRANTED;
+                if (permissionsGranted) {
+                    String telephoneNumber = ((EditText) findViewById(R.id.telephoneNumber)).getText().toString();
+                    // Checking the peer.
+                    SMSPeer peer = new SMSPeer(telephoneNumber);
```

Comment:
You are right!
I haven't thought about any kind of try-catch in my code... Especially because, in general, I have to be careful about using it, or I can ruin the variables' life cycle.

I'm gonna try to do it in next commit...
Thank you for the tip!

Reply:
I think I improved it now... What do you think about it?

Reply:
I think it's okay how you handled the exception.